app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,6 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_ingredients = RecipeIngredient.where(recipe: @recipe.id)
```

Comment:
This controller is off to a good start, but I'm not sure why we need to find all RecipeIngrdients by a specific Recipe's ID.... this reads to me like we're searching for data that we can already get by an association, like
```
@recipe.ingredients
```

In addition, using `.where` in a controller is considered an MVC infraction, since it's altering the query in a dynamic way - we're not sure what data we'll get back, so we would need to test this query to be sure it is correct. If you do need to use this query, it should be moved to a model method, tested, and then called here in the controller. But overall I don't think you need to do this specific query.